Source/UncompressedAudioSampleProvider.cpp

```diff
@@ -76,64 +79,74 @@ void UncompressedAudioSampleProvider::SetMediaEncodingProperties(AVSampleFormat
     if (outSampleFormat == AV_SAMPLE_FMT_S64)
         outSampleFormat = AV_SAMPLE_FMT_S32;
 
-    outChannels = inChannels;
-    outChannelLayout = inChannelLayout;
+    av_channel_layout_copy(&inChannelLayout, &channelLayout);
+    av_channel_layout_copy(&outChannelLayout, &channelLayout);
+
+    if (outChannelLayout.order != AV_CHANNEL_ORDER_NATIVE || !outChannelLayout.u.mask)
+    {
+        // TODO use av_channel_layout_retype once it becomes available in ffmpeg 7.0?!
```

Comment:
The API should be there now, we should use it to automatically convert to our desired layout.

Reply:
I got me here, completely missed this todo.

I don't think the first condition in that if should ever happen, because we are never setting the av_channel_order to anything other than native. Maybe this could happen due to filtering?

So that leaves the second part, at which point we force AV_CHANNEL_ORDER_NATIVE?